src/babashka/process.cljc

```diff
@@ -178,17 +178,28 @@
       (str/includes? "windows")))
 
 (defn- -program-resolver [{:keys [program dir]}]
-  ;; this should make life easier and not cause any bugs that weren't there previously
-  ;; on exception we just return the program as is
-  (try
-    (if (fs/relative? program)
-      (if-let [f (fs/which (if dir
-                             (-> (fs/file dir program) fs/absolutize)
-                             program))]
-        (str f)
-        program)
-      program)
-    (catch Throwable _ program)))
+  ;; Used by default:
+  ;; - for all program resolution when running on Windows
+  ;; - by `exec` on all OSes
+  ;; Default ProcessBuilder behaviour on macOS/Linux does this work naturally
+  ;; and is therefore unneeded.
+  ;;
+  ;; ProcessBuilder program resolution on Windows does not entirely match
+  ;; behaviour of CMD Shell nor PowerShell.
+  ;; Here we adapt resolution to match PowerShell (with the exclusion of resolving
+  ;; .ps1 scripts) which follows the same strategy as macOS/Linux.
+  (if-let [resolved (cond
+                      (fs/absolute? program)
+                      (fs/which program) ;; to resolve any extensions
+
+                      (fs/parent program)
+                      (fs/which (fs/file (or dir (fs/cwd))
+                                         program))
+
+                      :else
+                      (fs/which program))]
+    (-> resolved fs/absolutize fs/normalize str)
+    (throw (ex-info (str "Cannot resolve program: " program) {}))))
```

Comment:
I'd prefer leaving in the old behavior here and fall back to just `program` and let ProcessBuilder do the crashing for us, this is the same behavior we had before, so let's not change this.

Reply:
[ProcessBuilder on Windows will resolve `some-program` from cwd first then the PATH](https://github.com/lread/info-process-builder/blob/main/README.adoc#how-does-a-resolve-from-cwd). 
We don't want it to resolve to cwd first. 

I felt ProcessBuilder's behaviour on Windows is different enough from what we want and [broken enough](https://github.com/lread/info-process-builder/blob/main/README.adoc#how-does-aext-with-workdir-resolve-from-cwd-workdir-path) that it is clearer and less brittle for us to take over.